hri_safety_sense/src/VscProcess.cpp

```diff
@@ -36,186 +35,215 @@
 #include "hri_c_driver/VehicleInterface.h"
 #include "hri_c_driver/VehicleMessages.h"
 
+using namespace std::placeholders;
 using namespace hri_safety_sense;
-using namespace hri_safety_sense_srvs;
 
-VscProcess::VscProcess() :
-	myEStopState(0)
+namespace hri_safety_sense {
+
+#define THROTTLE(clock, duration, thing) \
+{ \
+  static rclcpp::Time _last_output_time(0, 0, clock->get_clock_type()); \
+  auto _now = clock->now(); \
+  if (_now - _last_output_time > duration) { \
+    _last_output_time = _now; \
+    thing; \
+  } \
+}
+
+VscProcess::VscProcess(rclcpp::NodeOptions &node_options) :
+  rclcpp::Node("VscProcess", node_options), myEStopState(0)
 {
-	ros::NodeHandle nh("~");
-	std::string serialPort = "/dev/ttyACM0";
-	if(nh.getParam("serial", serialPort)) {
-		ROS_INFO("Serial Port updated to:  %s",serialPort.c_str());
-	}
-
-	int  serialSpeed = 115200;
-	if(nh.getParam("serial_speed", serialSpeed)) {
-		ROS_INFO("Serial Port Speed updated to:  %i",serialSpeed);
-	}
-
-	std::string frameId = "/srcs";
-	if(nh.getParam("frame_id", frameId)) {
-		ROS_INFO("Frame ID updated to:  %s",frameId.c_str());
-	}
-
-	/* Open VSC Interface */
-	vscInterface = vsc_initialize(serialPort.c_str(),serialSpeed);
-	if (vscInterface == NULL) {
-		ROS_FATAL("Cannot open serial port! (%s, %i)",serialPort.c_str(),serialSpeed);
-	} else {
-		ROS_INFO("Connected to VSC on %s : %i",serialPort.c_str(),serialSpeed);
-	}
-
-	// Attempt to Set priority
-	bool  set_priority = false;
-	if(nh.getParam("set_priority", set_priority)) {
-		ROS_INFO("Set priority updated to:  %i",set_priority);
-	}
-
-	if(set_priority) {
-		if(setpriority(PRIO_PROCESS, 0, -19) == -1) {
-			ROS_ERROR("UNABLE TO SET PRIORITY OF PROCESS! (%i, %s)",errno,strerror(errno));
-		}
-	}
-
-	// Create Message Handlers
-	joystickHandler = new JoystickHandler(frameId);
-
-	// EStop callback
-	estopServ = rosNode.advertiseService("safety/service/send_emergency_stop", &VscProcess::EmergencyStop, this);
-
-	// KeyValue callbacks
-	keyValueServ = rosNode.advertiseService("safety/service/key_value", &VscProcess::KeyValue, this);
-	keyStringServ = rosNode.advertiseService("safety/service/key_string", &VscProcess::KeyString, this);
-
-	// Publish Emergency Stop Status
-	estopPub = rosNode.advertise<std_msgs::UInt32>("safety/emergency_stop", 10);
-
-	// Main Loop Timer Callback
-	mainLoopTimer = rosNode.createTimer(ros::Duration(1.0/VSC_INTERFACE_RATE), &VscProcess::processOneLoop, this);
-
-	// Init last time to now
-	lastDataRx = ros::Time::now();
-
-	// Clear all error counters
-	memset(&errorCounts, 0, sizeof(errorCounts));
+  std::string serialPort = "/dev/ttyACM0";
+  this->declare_parameter<std::string>("serial", serialPort);
+  if (this->get_parameter_or<std::string>("serial", serialPort, serialPort))
+    RCLCPP_INFO(this->get_logger(), "Serial Port updated to:  %s", serialPort.c_str());
```

Comment:
You can simplify this whole thing by using the return value from `declare_parameter`:

```
std::string serialPort = this->declare_parameter("serial", "/dev/ttyACM0");
RCLCPP_INFO(this->get_logger(), "Serial port set to: %s", serialPort.c_str());
```

The same goes for all of the declarations below.

Reply:
Done in 45091aa